### PR TITLE
iOS: Resolves #11119: Add iOS Dark Icon

### LINKED ISSUE #11119: Create the iOS 18 Dark icon
Ref: https://discourse.joplinapp.org/t/ios-18-enhancement-request-dark-icon/40330

Based on the provided Apple documentation, I guess it would be a black/dark grey icon with a blue "J" letter. Need to check if React Native supports this too.

### PR DESCRIPTION
**Pull request by itzTheMeow**
Resolves #11119.

Implements the dark icon for iOS 18, and converts the AppIcon asset to use the "Single Size" format rather than "All Sizes (Xcode 13)" format.
I tested the icon on the iPhone 16 simulator and it becomes dark and tinted properly.

### CODE REVIEW

**Comment by laurent22** [comment]
Thanks for the PR. I haven't reviewed yet but would you mind sharing screenshots of how it would be with this change?

**Comment by itzTheMeow** [comment]
simulator_screenshot_2469D6C8-7FD7-42DD-93CA-E8F175C43183
simulator_screenshot_D171843C-FF05-4ED1-B08C-CF276E444C14
simulator_screenshot_CA60C8E5-C9C3-424E-8595-32CB0CD2DCD5

File: packages/app-mobile/ios/Joplin/Joplin.entitlements [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
-	</array>
-</dict>
-</plist>
+	<dict>
+		<key>com.apple.security.application-groups</key>
+		<array>
+			<string>group.net.cozic.joplin</string>
+		</array>
+	</dict>
+</plist>
```
Comment (laurent22) [inline packages/app-mobile/ios/Joplin/Joplin.entitlements]: Could you please remove this whitespace change

File: packages/app-mobile/ios/Podfile.lock [hunk 2]
```diff
@@ -1776,4 +1776,4 @@ SPEC CHECKSUMS:
 
 PODFILE CHECKSUM: 11a2f8ebab99f816b8905858bff8a86a196b1f7e
 
-COCOAPODS: 1.16.1
+COCOAPODS: 1.16.2
```
Comment (laurent22) [inline packages/app-mobile/ios/Podfile.lock]: Also this I assume is not needed

File: packages/app-mobile/ios/ShareExtension/ShareExtension.entitlements [hunk 3]
```diff
@@ -1,10 +1,10 @@
...
-	</array>
-</dict>
-</plist>
+	<dict>
+		<key>com.apple.security.application-groups</key>
+		<array>
+			<string>group.net.cozic.joplin</string>
+		</array>
+	</dict>
+</plist>
```
Comment (laurent22) [inline packages/app-mobile/ios/ShareExtension/ShareExtension.entitlements]: Please remove

File: packages/tools/generate-images.ts [hunk 4]
```diff
@@ -1,7 +1,7 @@
-import { pathExists, readFile, writeFile, copyFile, readdir } from 'fs-extra';
+import { fileExtension } from '@joplin/lib/path-utils';
+import { copyFile, pathExists, readdir, readFile, writeFile } from 'fs-extra';
 import { dirname } from 'path';
 import { execCommand } from './tool-utils';
-import { fileExtension } from '@joplin/lib/path-utils';
 const md5File = require('md5-file');
```
Comment (laurent22) [inline packages/tools/generate-images.ts]: Also please restore this. We try to restrict the commits to just what's needed to make reviewing, reverting and merging easier

**Comment by laurent22** [comment]
Thanks for the update. I see that you removed the icons of various sizes - those are no longer needed, even for older devices?

**Comment by itzTheMeow** [comment]
From my research, the only devices that would be affected are iOS 12 and under, which the Joplin app doesn't support.
Xcode 14 release notes: https://developer.apple.com/documentation/xcode-release-notes/xcode-14-release-notes#Asset-Catalogs
Documentation for app icons: https://developer.apple.com/documentation/xcode/configuring-your-app-icon#Specify-app-icon-variations

Seems like Xcode added the white space changes, i'll revert.

**Comment by laurent22** [comment]
Thanks a lot for implementing this @itzTheMeow!

**Comment by laurent22** [comment]
@itzTheMeow, unfortunately I had to revert this change for now as it's not being accepted by Apple's servers.

The errors I'm getting are these ones:

> Asset validation failed
> 
> Missing Info.plist value. A value for the Info.plist key 'CFBundleIconName' is missing in the bundle 'net.cozic.joplin'. Apps built with iOS 11 or later SDK must supply app icons in an asset catalog and must also provide a value for this Info.plist key. For more information see http://help.apple.com/xcode/mac/current/#/dev10510b1f7. (ID: 6ee761d1-9848-4fb4-b586-1a14eaf8c59c)
> 
> Asset validation failed
> 
> Missing required icon file. The bundle does not contain an app icon for iPhone / iPod Touch of exactly '120x120' pixels, in .png format for iOS versions >= 10.0. To support older versions of iOS, the icon may be required in the bundle outside of an asset catalog. Make sure the Info.plist file includes appropriate entries referencing the file. See https://developer.apple.com/documentation/bundleresources/information_property_list/user_interface (ID: f28130a8-d88a-439d-b631-9718e6535350)
> 
> Asset validation failed
> 
> Missing required icon file. The bundle does not contain an app icon for iPad of exactly '167x167' pixels, in .png format for iOS versions supporting iPad Pro. To support older operating systems, the icon may be required in the bundle outside of an asset catalog. Make sure the Info.plist file includes appropriate entries referencing the file. See https://developer.apple.com/documentation/bundleresources/information_property_list/user_interface (ID: 487a277c-a069-4f59-96f6-f3f3d83400e0)
> 
> Asset validation failed
> 
> Missing required icon file. The bundle does not contain an app icon for iPad of exactly '152x152' pixels, in .png format for iOS versions >= 10.0. To support older operating systems, the icon may be required in the bundle outside of an asset catalog. Make sure the Info.plist file includes appropriate entries referencing the file. See https://developer.apple.com/documentation/bundleresources/information_property_list/user_interface (ID: b534a218-a9d7-4330-bbe3-35f09a6a4fa7)

And it's been reverted in edbb6137ea4d67e17d41be9ec77704b9f80cf444. Please let me know if you have any suggestions on how to fix this.

**Comment by itzTheMeow** [comment]
Just saw this, sorry. I'll take a look at it when i have time.

**Comment by itzTheMeow** [comment]
What is the build process used for submitting a new release? (xcode version, setup, etc)

**Pull request merged**